lib/dpp/isolation/IsolatedDpp.js

```diff
@@ -0,0 +1,220 @@
+const bootstrapIsolateFactory = require('./bootstrapIsolateFactory');
+const wrapDppCreateMethod = require('./wrapDppCreateMethod');
+const wrapDppValidateMethod = require('./wrapDppValidateMethod');
+const invokeFunctionFromIsolate = require('./invokeFunctionFromIsolate');
+
+const DPP_MODEL_TYPES = {
+  DATA_CONTRACT: 'dataContract',
+  DOCUMENT: 'document',
+  STATE_TRANSITION: 'stateTransition',
+  IDENTITY: 'identity',
+};
+
+class IsolatedDpp {
+  /**
+   *
+   * @param defaultExecutionOptions
+   * @param dpp
+   * @param snapshot
+   * @param dataProvider
+   * @param isolateOptions
+   * @returns {DashPlatformProtocol}
+   */
+  constructor(defaultExecutionOptions, dpp, snapshot, dataProvider, isolateOptions) {
+    const bootstrapIsolate = bootstrapIsolateFactory(snapshot, dataProvider, isolateOptions);
+
+    this.dataContract = {
```

Comment:
Let's extract facade initialization to `initializeFacades` as in original class

Reply:
Done